src/librustc_typeck/diagnostics.rs

```diff
@@ -1118,6 +1118,36 @@ fn main() {
 ```
 "##,
 
+E0102: r##"
+You hit this error because the compiler the compiler lacks information
+to determine a type for this variable. Erroneous code example:
+
+```
+fn main() {
+    let x: &_; // error: cannot determine a type for this local variable
```

Comment:
This won’t compile if we ever begin testing code samples in explanations. Please use something specific.